enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -82,13 +86,27 @@ private void dropAPIKeyDataFromAPIKeyHeader(RequestContext requestContext) throw
 
     private String getAPIKeyFromRequest(RequestContext requestContext) {
         Map<String, String> headers = requestContext.getHeaders();
-        return ConfigHolder.getInstance().getConfig().getApiKeyConfig().getApiKeyInternalHeaders().stream()
+        return getAPIKeyInternalHeaders(requestContext).stream()
                 .map(headers::get)
                 .filter(Objects::nonNull)
                 .findFirst()
                 .orElse(null);
     }
 
+    private List<String> getAPIKeyInternalHeaders(RequestContext requestContext) {
+        APIConfig apiConfig = requestContext.getMatchedAPI();
+        String configuredApiKeyHeader = apiConfig.getApiKeyHeader();
+        List<String> apiKeyInternalHeaders
+                = ConfigHolder.getInstance().getConfig().getApiKeyConfig().getApiKeyInternalHeaders();
+        if (StringUtils.isNotEmpty(configuredApiKeyHeader)) {
+            apiKeyInternalHeaders.add(configuredApiKeyHeader);
```

Comment:
Won't this add this item to the global list? I mean `ConfigHolder.getInstance().getConfig().getApiKeyConfig().getApiKeyInternalHeaders()` returns a pointer to the global list, and add() will add an item in that global list which will affect all the APIs.

Reply:
We should prioritize the `configuredApiKeyHeader`, right?

Reply:
Fixed